src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -1630,25 +1632,42 @@ void java_lang_Thread::set_name(oop java_thread, oop name) {
   java_thread->obj_field_put(_name_offset, name);
 }
 
+// Convenience macros for setting and getting Thread fields that
+// are actually stored in the FieldHolder object of the thread.
+// The FieldHolder can be null whilst a thread is attaching via JNI.
```

Comment:
Also the primordial thread. A JVMTI agent with the can_generate_early_vmstart capability sees the start phase before create_initial_thread and so many instrument some classes that are loaded early in the startup. Agents playing in this phase of startup are playing with fire of course.